csrc/host_ir/host_ir.cpp

```diff
@@ -48,42 +48,52 @@ bool HostUnit::sameAs(const Statement* other) const {
 
 PostOnStream::PostOnStream(
     IrBuilderPasskey passkey,
-    HostUnit* hu,
+    Expr* host_op,
     std::vector<Val*> inputs,
     std::vector<Val*> outputs)
-    : Expr(passkey, std::move(inputs), std::move(outputs), {hu}) {
+    : Expr(passkey, std::move(inputs), std::move(outputs), {host_op}) {
   NVF_ERROR(passkey.ir_container_->isA<hir::HostIrContainer>()); // NOLINT
-  NVF_ERROR(this->inputs().size() == hu->fusion_to_execute()->inputs().size());
-  NVF_ERROR(
-      this->outputs().size() == hu->fusion_to_execute()->outputs().size());
-  // TODO: harden the assert checks with smth like
-  // for (int i : c10::irange(inputs.size())) {
-  //     // NVF_ERROR(inputs.at(i)->sameAs(executable_fusion->inputs().at(i)));
-  // }
-  // for (int i : c10::irange(outputs.size())) {
-  //     //
-  //     NVF_ERROR(outputs.at(i)->sameAs(executable_fusion->outputs().at(i)));
-  // }
+  if (host_op->isA<HostUnit>()) {
```

Comment:
Could you add some more info on what `host_op` is allowed to be? IIUC, at this point, it should be either `HostUnit` or `Communication`?

Reply:
> IIUC, at this point, it should be either HostUnit or Communication?

correct!

> Could you add some more info on what host_op is allowed to be?

I'll add a else branch with a `NVF_ERROR(false,...)` to assert that. I can also add some comment about that where `PostOnStream` is defined

Reply:
Weird, I needed to add surrounding parenthis for it to work...